### PR TITLE
planner/core: migrate `physical_plan_test.go` to testify - batch 1

### PR DESCRIPTION
**Pull request by johnhaxx7**
### What problem does this PR solve?

Issue Number: ref #28409

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```
Hi, this is the initial PR for this migration. 
This file is too large, please feel free to let me know if you have better refactor ideas.

### CODE REVIEW

File: planner/core/physical_plan_test.go [hunk 1]
```diff
@@ -590,7 +531,7 @@ func (s *testPlanSuite) TestINMJHint(c *C) {
 		dom.Close()
 		store.Close()
 	}()
-	tk := testkit.NewTestKit(c, store)
+	tk := oldtestkit.NewTestKit(c, store)
```
Comment (johnhaxx7) [inline planner/core/physical_plan_test.go:534]: below this line, all the changes in this file is just renaming testkit to oldtestkit
oldtestkit pkg will be removed after migration finish

File: planner/core/prepare_test.go [hunk 2]
```diff
@@ -28,7 +28,9 @@ import (
 	"github.com/pingcap/tidb/infoschema"
 	"github.com/pingcap/tidb/kv"
 	"github.com/pingcap/tidb/metrics"
+	"github.com/pingcap/tidb/parser"
```
Comment (johnhaxx7) [inline planner/core/prepare_test.go:31]: all the change from this file are moved from physical_plan_test.go

File: planner/core/physical_plan_test.go [hunk 3]
```diff
@@ -35,100 +36,62 @@ import (
...
-		store.Close()
-	}()
-	se, err := session.CreateSession4Test(store)
-	c.Assert(err, IsNil)
-	_, err = se.Execute(context.Background(), "use test")
-	c.Assert(err, IsNil)
-	_, err = se.Execute(context.Background(), "set tidb_opt_limit_push_down_threshold=0")
-	c.Assert(err, IsNil)
+func TestDAGPlanBuilderSimpleCase(t *testing.T) {
+	defer testleak.AfterTestT(t)()
```
Comment (tisonkun) [inline planner/core/physical_plan_test.go:56]: goroutine leaks are already covered by `main_test.go`.

Please refer to other migrated tests under `planner/core` package for examples.

Comment (tisonkun) [reply planner/core/physical_plan_test.go:56]: e.g. https://github.com/pingcap/tidb/pull/32510

File: planner/core/physical_plan_test.go [hunk 4]
```diff
@@ -19,6 +19,7 @@ import (
 	"fmt"
 	"math"
 	"strings"
+	"testing"
 
 	. "github.com/pingcap/check"
```
Comment (tisonkun) [inline planner/core/physical_plan_test.go:24]: A correct migration should be able to get rid of this dependency.

File: planner/core/prepare_test.go (hunk 2 above)
Comment (tisonkun) [reply planner/core/prepare_test.go:31]: `prepare_test.go` has been migrated, please rebase on the master.

**Comment by tisonkun** [comment]
I'm going to takeover this issue and do it at once. You're welcome to give the following PR a review :)

**Pull request closed without merging**